lusidtools/cocoon/utilities.py

```diff
@@ -314,7 +327,24 @@ def set_attributes_recursive(
         return None
 
     # Create an instance of and populate the model object
-    return model_object(**obj_init_values)
+    instance = model_object(**obj_init_values)
+
+    # Support for polymorphism, we can identify these `abstract` classes by the existence of the below
+    if getattr(instance, "discriminator"):
+        discriminator = getattr(instance, getattr(instance, "discriminator"))
+        try:
+            actual_class = model_object.discriminator_value_class_map[discriminator]
+        except TypeError:
+            print(
+                f"Invalid discriminator object {discriminator} was passed. "
+                f"Please ensure is one of: {model_object.discriminator_value_class_map}"
+            )
```

Comment:
Great checking, can we please use `logging.error` instead of `print`?

Reply:
Certainly!